src/cli/alcotest_cli.ml

```diff
@@ -0,0 +1,246 @@
+(*
+ * Copyright (c) 2013-2016 Thomas Gazagnaire <thomas@gazagnaire.org>
+ * Copyright (c) 2019 Craig Ferguson <me@craigfe.io>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+open Cmdliner
+open Astring
+open Alcotest
+module IntSet = Core.IntSet
+
+module type S = sig
+  include Core.S
+
+  val run :
+    (?argv:string array -> string -> unit test list -> return) with_options
+
+  val run_with_args :
+    (?argv:string array ->
+    string ->
+    'a Cmdliner.Term.t ->
+    'a test list ->
+    return)
+    with_options
+end
+
+module Make (M : Monad.S) : S with type return = unit M.t = struct
+  module C = Core.Make (M)
+  include C
+
+  let set_color style_renderer = Fmt_tty.setup_std_outputs ?style_renderer ()
+
+  let set_color = Term.(const set_color $ Fmt_cli.style_renderer ())
+
+  (* arg_* types used as an ad-hoc "named argument" feature to prevent flags
+     from being re-ordered when passing to Core.run. *)
+  type arg_verbose = Verbose of bool
+
+  type arg_compact = Compact of bool
+
+  type arg_show_errors = Show_errors of bool
+
+  type arg_quick_only = Quick_only of bool
+
+  type arg_json = Json of bool
+
+  type arg_log_dir = Log_dir of string
+
+  type arg_test_filter =
+    | Test_filter of (Re.re option * IntSet.t option) option
+
+  type runtime_options = {
+    verbose : bool;
+    compact : bool;
+    show_errors : bool;
+    quick_only : bool;
+    json : bool;
+    log_dir : string option;
+  }
+
+  let v_runtime_flags ~defaults (Verbose verbose) (Compact compact)
+      (Show_errors show_errors) (Quick_only quick_only) (Json json)
+      (Log_dir log_dir) =
```

Comment:
Nice trick ! Would that work with polymorphic variants to avoid defining the types with dummy names ?

Reply:
Good idea :slightly_smiling_face: Done.